Omit the directory and show the distance

Herbie34 opened the pull request:
I've completed all the quests! The game balance was perfect, and I really enjoyed it.

By the way, I was wondering if it's fine to omit the display of the current directory.
Also, I thought it might be a good idea to show the distance. What do you think?
(I don't have any experience with Rust development, so I apologize if anything is incorrect.)

Herbie34 committed:
Omit the directory and show the distance

Herbie34 commented:
On Linux, components[0] is just '/', so it might be better to also add components[1].

facundoolano commented:
Can you show some before/after examples of this? 

Perhaps a better option than the ... would be to show a compact version of the path segments, eg. my shell prompt does something like this: `~/d/f/o/t/assets $`

Herbie34 commented:
Before(use `dn` and `rpg status`):
 warrior[100]@~/dungeon/1/2/3/4/5
 ...

After(add components[1]):
 warrior[100]@~/dungeon/.../5[6]
 ...

It might be a good idea to display only the first character of each directory, but I thought it might get quite long when you go deep into directories for quests...

facundoolano commented:
Gotcha, let me think about this for a bit.

Herbie34 committed:
Add 2nd directory

On Linux, components[0] is just '/',
so it might be better to also add components[1].

Herbie34 committed:
Change the directory display according to the environment variable

Herbie34 committed:
Merge branch 'main' into omit_directory
